File: backend/MedicalSystem/Models/DoctorRating.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedicalSystem.Models
{
    public class DoctorRating
    {
        [Key]
        public int PID { get; set; }
        [Key]
        public int DID { get; set; }
        public int Rating { get; set; }

        //[ForeignKey("DID")]
        //[InverseProperty("DoctorRatings")]
        //public virtual Doctor DIDNavigation { get; set; }
        //[ForeignKey("PID")]
        //[InverseProperty("DoctorRatings")]
        //public virtual Patient PIDNavigation { get; set; }


    }
}

File: Frontend/src/app/_Models/doctor-rating.ts
import { Doctor } from "./doctor";
import { Patient } from "./patient";

export class DoctorRating {
    constructor(
                public pid:number=Number(),
                public did:number=Number(),
                public VisitNumber:number=0,
                public Rating:number=Number(),
                public didNavigation:Doctor|null=null,
                public pidNavigation:Patient|null=null
            )
    {}
}